src/synchronicity/type_stubs.py

```diff
@@ -843,8 +844,13 @@ def _get_function_source(
             self.imports.add("typing_extensions")
             maybe_decorators = f"{signature_indent}@typing_extensions.dataclass_transform({args})\n"
 
+        def is_async(func):
+            if hasattr(func, "__wrapped__") and not hasattr(func, SYNCHRONIZER_ATTR):
```

Comment:
I'd change the second half to `and getattr(func, TARGET_INTERFACE_ATTR, None) != Interface.BLOCKING`, which I think should work too, but it's a bit more "correct"

Difference being that if it stops the recursion at any synchronizer, it would also stop if it reaches a `.aio` method, which may or may not be a coroutinefunction depending on some synchronicity internals that we might want to change.

And maybe move this as a utility function into the async_wrap module

Reply:
👍